fitlins/cli/run.py

```diff
@@ -170,18 +168,53 @@ def run_fitlins(argv=None):
         pipeline_name += '_' + opts.derivative_label
     deriv_dir = op.join(opts.output_dir, pipeline_name)
     os.makedirs(deriv_dir, exist_ok=True)
-
     bids.write_derivative_description(opts.bids_dir, deriv_dir)
 
     work_dir = mkdtemp() if opts.work_dir is None else opts.work_dir
 
+    # Go ahead and initialize the layout database
+    if opts.database_path is None:
+        database_path = Path(work_dir) / 'dbcache'
+        reset_database = True
+        make_layout = True
+    elif Path(opts.database_path).exists():
+        layout = BIDSLayout.load(opts.database_path)
```

Comment:
It might be better to let pybids handle some of this logic.

That is, is path exists, let pybids decide to load the database from it. 
The problem with the current approach is that if someone passes a valid db path, but then changes the CLIs arguments (so that they would actually want to re-index), no error will be thrown. So their CLI options are being silently ignored in favor for the indexing options in the db_path. 

I rather let pybids crash if the options mismatch. Also simplifies this logic.

Reply:
:+1: